core/src/main/scala/kafka/tools/ConsoleConsumer.scala

```diff
@@ -138,21 +138,25 @@ object ConsoleConsumer extends Logging {
             throw e
           }
       }
-      checkErr(formatter)
+      if (checkErr(output, formatter)) {
+        // Consumer will be closed
+        return
```

Comment:
Hmm, introducing a new message type doesn't seem worth it, and I see now that if we left it uncaught (which otherwise has the same effect of forcing us to exit) we would log an extra "Unknown exception" message at the top-level exception handler.

Let's leave it as is, although I agree that the `formatter` could just be always closed in a `finally` block. I guess nobody has noticed that gap because most formatters don't even need to implement `close()`.